mitmproxy/proxy/protocol2/websocket.py

```diff
@@ -16,14 +19,104 @@ def __init__(self, context: ClientServerContext, handshake_flow: http.HTTPFlow):
         super().__init__(context)
         self.flow = websocket.WebSocketFlow(context.client, context.server, handshake_flow)
         assert context.server.connected
+        self.client_frame_buffer = []
+        self.server_frame_buffer = []
+        self.extension = None
+        self.make_extension()
+        self.client_conn = wsconn.WSConnection(wsconn.SERVER, wsconn.ConnectionState.OPEN,
+                                               extensions=[self.extension] if self.extension else None)
+
+        self.server_conn = wsconn.WSConnection(wsconn.CLIENT, wsconn.ConnectionState.OPEN,
+                                               extensions=[self.extension] if self.extension else None)
+
+    def make_extension(self):
```

Comment:
We can make our code simpler by making this function side-effect free, i.e. let's provide `server_extensions` as an argument and return the extension here instead of storing it as `self.extension`. In principle it's a `@staticmethod` then.